docs/http-vertx.md

```diff
@@ -0,0 +1,102 @@
+---
+title: CloudEvents HTTP Vert.x
+nav_order: 5
+---
+
+# HTTP Protocol Binding for Eclipse Vert.x
+
+[![Javadocs](http://www.javadoc.io/badge/io.cloudevents/cloudevents-http-vertx.svg?color=green)](http://www.javadoc.io/doc/io.cloudevents/cloudevents-http-vertx)
+
+For Maven based projects, use the following to configure the CloudEvents Vertx
+HTTP Transport:
+
+```xml
+<dependency>
+    <groupId>io.cloudevents</groupId>
+    <artifactId>cloudevents-http-vertx</artifactId>
+    <version>2.0.0.RC1</version>
+</dependency>
+```
+
+## Receiving CloudEvents
+
+Assuming you have in classpath [`cloudevents-json-jackson`](json-jackson.md),
+below is a sample on how to read and write CloudEvents:
+
+```java
+
+import io.cloudevents.http.vertx.VertxMessageFactory;
+import io.cloudevents.core.message.StructuredMessageReader;
+import io.cloudevents.CloudEvent;
+import io.vertx.core.AbstractVerticle;
+
+public class CloudEventServerVerticle extends AbstractVerticle {
+
+  public void start() {
+    vertx.createHttpServer()
+      .requestHandler(req -> {
+        VertxMessageFactory.createReader(req)
+          .onComplete(result -> {
+            // If decoding succeeded, we should write the event back
+            if (result.succeeded()) {
+              CloudEvent event = result.result().toEvent();
+              // Echo the message, as structured mode
+              VertxMessageFactory
+                .createWriter(req.response())
+                .writeStructured(event, "application/cloudevents+json");
+            }
+            req.response().setStatusCode(500).end();
+          });
```

Comment:
Can you use onSucceed / onFailure? You also need to return at the end of the if block if you want to keep the onComplete.